src/dialogs/copyPasteOwnership.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { AlertVariant } from "@patternfly/react-core/dist/esm/components/Alert";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect";
+import { Modal, ModalVariant } from "@patternfly/react-core/dist/esm/components/Modal";
+
+import cockpit from 'cockpit';
+import type { Dialogs, DialogResult } from 'dialogs';
+import { superuser } from 'superuser';
+
+import { useFilesContext } from '../app.tsx';
+import type { ClipboardInfo } from '../app.tsx';
+import { get_owner_candidates } from '../ownership.tsx';
+
+const _ = cockpit.gettext;
+
+async function pasteAsOwner(clipboard: ClipboardInfo, dstPath: string, owner: string,
+    addAlert: (title: string, variant: AlertVariant, key: string, detail?: string) => void) {
+    try {
+        await cockpit.spawn([
+            "cp",
+            "--recursive",
+            ...clipboard.files.map(file => clipboard.path + file.name),
+            dstPath
+        ], { superuser: "try" });
+
+        await cockpit.spawn([
+            "chown",
+            "--recursive",
+            owner,
+            ...clipboard.files.map(file => dstPath + file.name),
+        ], { superuser: "try" });
+    } catch (err) {
+        const e = err as cockpit.BasicError;
```

Comment:
Error handling is a bit tricky here. It could theoretically happen that the `cp` works, but the `chown` fails (not very likely, but think of "target user is unknown or invalid"). Or that the `cp` fails but already copied half of the files. So similar to #845 this needs to do some cleanup here -- best to remove everything you may have copied already.

What pains me is that upload, drag and drop, and now here pasting essentially do the same things and all have the same issues. This really could use some factorization of the cp/chown/chmod bits which all three reuse.